src/common/api_utils.cpp

```diff
@@ -44,80 +43,39 @@ string APIUtils::GetAwsService(const string &host) {
 }
 
 string APIUtils::GetRequestAws(ClientContext &context, IRCEndpointBuilder endpoint_builder, const string &secret_name) {
-	auto clientConfig = make_uniq<Aws::Client::ClientConfiguration>();
-
-	if (!SELECTED_CURL_CERT_PATH.empty()) {
-		clientConfig->caFile = SELECTED_CURL_CERT_PATH;
-	}
-
-	std::shared_ptr<Aws::Http::HttpClientFactory> MyClientFactory;
-	std::shared_ptr<Aws::Http::HttpClient> MyHttpClient;
-
-	MyHttpClient = Aws::Http::CreateHttpClient(*clientConfig);
-	Aws::Http::URI uri;
+	AWSInput aws_input;
```

Comment:
I'm wondering if we even need this code in `APIUtils::GetRequestAWS`? The logic seems very AWS specific and I don't mind if it goes int the AWSInput class somewhere